source/GameData.cpp

```diff
@@ -830,6 +831,16 @@ void GameData::LoadSources()
 
 		icon->ValidateFrames();
 		spriteQueue.Add(icon);
+
+		// Enabling plugins is the default preference.
+		if(Plugins::IsEnabled(name))
+		{
+			Plugins::Set(name, true);
+			++it;
+			continue;
+		}
+		// Remove sources for user-disabled plugins while preserving about.txt in preferences.
+		it = sources.erase(it);
```

Comment:
Instead of removing a source after the fact, don't add the source to the list in the first place.

Reply:
This is a catch 22.  I need to walk the sources to get the name.  I also need the about.txt to load for disabled plugins.  If this information isn't loaded then we could have a blank entry for disabled plugins.

I opted for this method because having the about.txt information for disabled plugins is user friendly.

Reply:
I'm going to keep this in since I want plugin preferences to show icon and `about.txt` for plugins.

Reply:
You can separate this: Make a new function at the top of the file that simply loads a given plugin. Then, when traversing the directory looking for plugins, call that new function and then only add the plugin to the source list if it is enabled.

Reply:
Are we trying to avoid popping items off of a list?  The list is rarely above 20 plugins/items (overflows plugin UI).  Popping items out of a list is a cheap operation adjusting pointers.

Are you suggesting deleting this icon/about loop entirely in favor of a function to be called inside of the other global/local loops?

Reply:
Should it be a namespaced function or private function?

Reply:
I pushed 0bad9f0557d12aa4a71f1bea1f023bc22c9ae933 to address your feedback.  @quyykk if you're satisfied I will defer to you to resolve this feedback by clicking the button since you've already unresolved it before.